src/main/java/ee/carlrobert/llm/client/watsonx/WatsonxAuthenticator.java

```
@@ -0,0 +1,128 @@
+    authParams.put("api_key", apiKey);
+
+    String authParamsStr = "";
+    try {
+      authParamsStr = mapper.writeValueAsString(authParams);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    MediaType mediaType = MediaType.parse("application/json");
+    RequestBody body = RequestBody.create(mediaType, authParamsStr);
+    // TODO add support for IAM endpoint v1/auth/identitytoken
+    this.request = new Request.Builder()
+        .url(host
+            + "/icp4d-api/v1/authorize")
+        .method("POST", body)
+        .addHeader("Content-Type", "application/json")
+        .build();
+
+    this.expiryRequest = new Request.Builder()
+        .url(host + "/usermgmt/v1/user/tokenExpiry")
+        .addHeader("Accept", "application/json")
+        .addHeader("Authorization", "Bearer " + this.bearerToken.getAccessToken())
+        .build();
+
+    try {
+      Response response = client.newCall(request).execute();
+      this.bearerToken = OBJECT_MAPPER.readValue(response.body().string(),
+          IBMAuthBearerToken.class);
+
+      Response expiry = client.newCall(request).execute();
+      this.bearerToken.setExpiration(
+          OBJECT_MAPPER.readValue(expiry.body().string(), IBMAuthTokenExpiry.class)
+              .getExpiry());
+
+    } catch (IOException e) {
+      System.out.println(e);
+    }
+  }
+
+  private void generateNewBearerToken() {
+    try {
+      Response response = client.newCall(request).execute();
+      this.bearerToken = OBJECT_MAPPER.readValue(response.body().string(),
+          IBMAuthBearerToken.class);
+      if (this.bearerToken.getExpiration() == null) {
+        Response expiry = client.newCall(expiryRequest).execute();
+        this.bearerToken.setExpiration(
+            OBJECT_MAPPER.readValue(expiry.body().string(), IBMAuthTokenExpiry.class)
+                .getExpiry());
+      }
+    } catch (IOException e) {
+      System.out.println(e);
```

Please re-throw the exception with meaningful message